<img src="https://www.fao.org/images/corporatelibraries/fao-logo/fao-logo-fr.svg?sfvrsn=f64522b4_31"  />

### Première Partie - Importation des librairies et préparation des dataframes


### I. Importation des librairies

In [77]:
# Importons la librairie pandas
import pandas as pd

### II. Lecture des fichiers et preparation des dataframes

#### II.1.  Lecture des fichiers

#### II.1.1. Population

In [78]:
# Récupérons notre fichier csv dans une variable
df_population = pd.read_csv('population.csv')

#### II.1.2. Disponibilité alimentaire

In [79]:
# Récupérons notre fichier csv dans une variable
df_dispo_alim = pd.read_csv('dispo_alimentaire.csv')

#### II.1.3. Sous nutrition

In [80]:
# Récupérons notre fichier csv dans une variable
df_sous_nutrition = pd.read_csv('sous_nutrition.csv')

#### II.1.3. Aide alimentaire

In [81]:
# Récupérons notre fichier csv dans une variable
df_aide_alim = pd.read_csv('aide_alimentaire.csv')

#### II.2 Preparation des dataframes

#### II.2.1 Population

In [82]:
# Affichons les 5 premières entrées pour avoir un aperçu de notre dataframe
df_population.head()

Zone  Année     Valeur
0  Afghanistan   2013  32269.589
1  Afghanistan   2014  33370.794
2  Afghanistan   2015  34413.603
3  Afghanistan   2016  35383.032
4  Afghanistan   2017  36296.113

In [83]:
# Affichons les 5 dernières entrées pour avoir un aperçu de notre dataframe
df_population.tail()

Zone  Année     Valeur
1411  Zimbabwe   2014  13586.707
1412  Zimbabwe   2015  13814.629
1413  Zimbabwe   2016  14030.331
1414  Zimbabwe   2017  14236.595
1415  Zimbabwe   2018  14438.802

In [84]:
# Affichons le nombre de lignes et de colonnes
df_population.shape

(1416, 3)

In [85]:
# Vérifions le type de données de notre dataframe
df_population.dtypes

Zone       object
Année       int64
Valeur    float64
dtype: object

In [86]:
#Recherchons des valeurs manquantes
df_population.isnull().sum()

Zone      0
Année     0
Valeur    0
dtype: int64

In [87]:
#Renommons le titre des colonnes 'zone' et 'valeur'
df_population.rename(columns={'Zone':'Pays','Valeur':'Nombre_habitants'},inplace=True)

In [553]:
# Affichons les colonnes de la dataframe 'df_population'
df_population.columns

Index(['Pays', 'Année', 'Nombre_habitants'], dtype='object')

In [89]:
# Modifions les valeurs Nombre_habitants en milliers d'habitants
df_population['Nombre_habitants']=df_population['Nombre_habitants']*1000

In [90]:
df_population['Nombre_habitants']

0       32269589.0
1       33370794.0
2       34413603.0
3       35383032.0
4       36296113.0
           ...    
1411    13586707.0
1412    13814629.0
1413    14030331.0
1414    14236595.0
1415    14438802.0
Name: Nombre_habitants, Length: 1416, dtype: float64

In [91]:
# Modifions le type de ma colonne en int
df_population['Nombre_habitants'] = df_population['Nombre_habitants'].astype(int)

In [92]:
df_population['Nombre_habitants'].dtypes

dtype('int32')

In [93]:
df_population['Nombre_habitants']

0       32269589
1       33370794
2       34413603
3       35383032
4       36296113
          ...   
1411    13586707
1412    13814629
1413    14030331
1414    14236595
1415    14438802
Name: Nombre_habitants, Length: 1416, dtype: int32

In [94]:
# Création de notre dataframe Population pour l'année 2017
df_population[df_population.Année==2017]  

Pays  Année  Nombre_habitants
4                                Afghanistan   2017          36296113
10                            Afrique du Sud   2017          57009756
16                                   Albanie   2017           2884169
22                                   Algérie   2017          41389189
28                                 Allemagne   2017          82658409
...                                      ...    ...               ...
1390  Venezuela (République bolivarienne du)   2017          29402484
1396                                Viet Nam   2017          94600648
1402                                   Yémen   2017          27834819
1408                                  Zambie   2017          16853599
1414                                Zimbabwe   2017          14236595

[236 rows x 3 columns]

In [95]:
df_population_2017 = df_population[df_population.Année==2017] 

In [96]:
df_population_2017.Nombre_habitants.sum()

7548134097

#### II.2.2. Disponibilité alimentaire

In [97]:
# Affichons les 5 premières entrées pour avoir un aperçu de notre dataframe
df_dispo_alim.head()

Zone                Produit   Origine  Aliments pour animaux  \
0  Afghanistan       Abats Comestible   animale                    NaN   
1  Afghanistan        Agrumes, Autres  vegetale                    NaN   
2  Afghanistan  Aliments pour enfants  vegetale                    NaN   
3  Afghanistan                 Ananas  vegetale                    NaN   
4  Afghanistan                Bananes  vegetale                    NaN   

   Autres Utilisations  Disponibilité alimentaire (Kcal/personne/jour)  \
0                  NaN                                             5.0   
1                  NaN                                             1.0   
2                  NaN                                             1.0   
3                  NaN                                             0.0   
4                  NaN                                             4.0   

   Disponibilité alimentaire en quantité (kg/personne/an)  \
0                                               1.72        
1                                               1.29        
2                                               0.06        
3                                               0.00        
4                                               2.70        

   Disponibilité de matière grasse en quantité (g/personne/jour)  \
0                                               0.20               
1                                               0.01               
2                                               0.01               
3                                                NaN               
4                                               0.02               

   Disponibilité de protéines en quantité (g/personne/jour)  \
0                                               0.77          
1                                               0.02          
2                                               0.03          
3                                                NaN          
4                                               0.05          

   Disponibilité intérieure  Exportations - Quantité  Importations - Quantité  \
0                      53.0                      NaN                      NaN   
1                      41.0                      2.0                     40.0   
2                       2.0                      NaN                      2.0   
3                       0.0                      NaN                      0.0   
4                      82.0                      NaN                     82.0   

   Nourriture  Pertes  Production  Semences  Traitement  Variation de stock  
0        53.0     NaN        53.0       NaN         NaN                 NaN  
1        39.0     2.0         3.0       NaN         NaN                 NaN  
2         2.0     NaN         NaN       NaN         NaN                 NaN  
3         0.0     NaN         NaN       NaN         NaN                 NaN  
4        82.0     NaN         NaN       NaN         NaN                 NaN

In [98]:
# Affichons le nombre de lignes et de colonnes
df_dispo_alim.shape

(15605, 18)

In [99]:
# Vérifions le type de données de notre dataframe
df_dispo_alim.dtypes

Zone                                                              object
Produit                                                           object
Origine                                                           object
Aliments pour animaux                                            float64
Autres Utilisations                                              float64
Disponibilité alimentaire (Kcal/personne/jour)                   float64
Disponibilité alimentaire en quantité (kg/personne/an)           float64
Disponibilité de matière grasse en quantité (g/personne/jour)    float64
Disponibilité de protéines en quantité (g/personne/jour)         float64
Disponibilité intérieure                                         float64
Exportations - Quantité                                          float64
Importations - Quantité                                          float64
Nourriture                                                       float64
Pertes                                             

In [100]:
# Recherchons les valeurs manquantes
df_dispo_alim.isnull().sum()

Zone                                                                 0
Produit                                                              0
Origine                                                              0
Aliments pour animaux                                            12885
Autres Utilisations                                              10109
Disponibilité alimentaire (Kcal/personne/jour)                    1364
Disponibilité alimentaire en quantité (kg/personne/an)            1590
Disponibilité de matière grasse en quantité (g/personne/jour)     3811
Disponibilité de protéines en quantité (g/personne/jour)          4044
Disponibilité intérieure                                           223
Exportations - Quantité                                           3379
Importations - Quantité                                            753
Nourriture                                                        1590
Pertes                                                           11327
Produc

In [101]:
# Remplaçons  mes valeurs nulles par 0 et créeons une nouvelle variable pour conserver la variable de départ
df_dispon_alim = df_dispo_alim.fillna(0)

In [102]:
# Affichons les 5 premières entrées pour avoir un aperçu de notre dataframe
df_dispon_alim.head()

Zone                Produit   Origine  Aliments pour animaux  \
0  Afghanistan       Abats Comestible   animale                    0.0   
1  Afghanistan        Agrumes, Autres  vegetale                    0.0   
2  Afghanistan  Aliments pour enfants  vegetale                    0.0   
3  Afghanistan                 Ananas  vegetale                    0.0   
4  Afghanistan                Bananes  vegetale                    0.0   

   Autres Utilisations  Disponibilité alimentaire (Kcal/personne/jour)  \
0                  0.0                                             5.0   
1                  0.0                                             1.0   
2                  0.0                                             1.0   
3                  0.0                                             0.0   
4                  0.0                                             4.0   

   Disponibilité alimentaire en quantité (kg/personne/an)  \
0                                               1.72        
1                                               1.29        
2                                               0.06        
3                                               0.00        
4                                               2.70        

   Disponibilité de matière grasse en quantité (g/personne/jour)  \
0                                               0.20               
1                                               0.01               
2                                               0.01               
3                                               0.00               
4                                               0.02               

   Disponibilité de protéines en quantité (g/personne/jour)  \
0                                               0.77          
1                                               0.02          
2                                               0.03          
3                                               0.00          
4                                               0.05          

   Disponibilité intérieure  Exportations - Quantité  Importations - Quantité  \
0                      53.0                      0.0                      0.0   
1                      41.0                      2.0                     40.0   
2                       2.0                      0.0                      2.0   
3                       0.0                      0.0                      0.0   
4                      82.0                      0.0                     82.0   

   Nourriture  Pertes  Production  Semences  Traitement  Variation de stock  
0        53.0     0.0        53.0       0.0         0.0                 0.0  
1        39.0     2.0         3.0       0.0         0.0                 0.0  
2         2.0     0.0         0.0       0.0         0.0                 0.0  
3         0.0     0.0         0.0       0.0         0.0                 0.0  
4        82.0     0.0         0.0       0.0         0.0                 0.0

In [103]:
# Modifions les valeurs disponibilité alimentaire en milliers de tonnes
df_dispon_alim[['Aliments pour animaux','Autres Utilisations','Disponibilité intérieure','Exportations - Quantité','Importations - Quantité','Nourriture','Pertes','Production','Semences','Traitement']] = df_dispon_alim[['Aliments pour animaux','Autres Utilisations','Disponibilité intérieure','Exportations - Quantité','Importations - Quantité','Nourriture','Pertes','Production','Semences','Traitement']]*1000000

In [104]:
# Vérifions le changement de nos valeurs en milliers de tonnes
df_dispon_alim[['Disponibilité intérieure','Exportations - Quantité','Importations - Quantité','Nourriture','Production',]].head()

Disponibilité intérieure  Exportations - Quantité  Importations - Quantité  \
0                53000000.0                      0.0                      0.0   
1                41000000.0                2000000.0               40000000.0   
2                 2000000.0                      0.0                2000000.0   
3                       0.0                      0.0                      0.0   
4                82000000.0                      0.0               82000000.0   

   Nourriture  Production  
0  53000000.0  53000000.0  
1  39000000.0   3000000.0  
2   2000000.0         0.0  
3         0.0         0.0  
4  82000000.0         0.0

In [105]:
# Renommons le titre de ma colonne'zone'
df_dispon_alim.rename(columns={'Zone':'Pays','Aliments pour animaux':'Aliments_Animaux','Autres Utilisations':'Autres_utilisations','Disponibilité alimentaire (Kcal/personne/jour)':'Dispo_alim_kcal','Disponibilité alimentaire en quantité (kg/personne/an)':'Dispo_alim_qte','Disponibilité de matière grasse en quantité (g/personne/jour)':'Dispo_mat_gras_qte','Disponibilité de protéines en quantité (g/personne/jour)':'Dispo_prot_qte','Disponibilité intérieure':'Dispo_inter','Exportations - Quantité':'Export_qte','Importations - Quantité':'Import_qte','Variation de stock':'Var_stock'},inplace=True)

In [106]:
# Affichons les colonnes de la dataframe
df_dispon_alim.columns

Index(['Pays', 'Produit', 'Origine', 'Aliments_Animaux', 'Autres_utilisations',
       'Dispo_alim_kcal', 'Dispo_alim_qte', 'Dispo_mat_gras_qte',
       'Dispo_prot_qte', 'Dispo_inter', 'Export_qte', 'Import_qte',
       'Nourriture', 'Pertes', 'Production', 'Semences', 'Traitement',
       'Var_stock'],
      dtype='object')

#### II.2.3. Sous nutrition

In [107]:
# Affichons les 5 premières entrées pour avoir un aperçu de notre dataframe
df_sous_nutrition.head()

Zone      Année Valeur
0  Afghanistan  2012-2014    8.6
1  Afghanistan  2013-2015    8.8
2  Afghanistan  2014-2016    8.9
3  Afghanistan  2015-2017    9.7
4  Afghanistan  2016-2018   10.5

In [108]:
# Affichons le nombre de lignes et de colonnes
df_sous_nutrition.shape

(1218, 3)

In [109]:
# Vérifions le type de données de notre dataframe
df_sous_nutrition.dtypes

Zone      object
Année     object
Valeur    object
dtype: object

In [110]:
# Renommons le titre de la colonne'zone' et 'valeur'
df_sous_nutrition.rename(columns={'Zone':'Pays','Valeur':'Nombre_personnes'},inplace=True)

In [111]:
df_sous_nutrition.columns

Index(['Pays', 'Année', 'Nombre_personnes'], dtype='object')

In [112]:
# Recherchons les valeurs manquantes
df_sous_nutrition.isnull().sum()

Pays                  0
Année                 0
Nombre_personnes    594
dtype: int64

In [117]:
# Remplacons les valeurs nulles par 0
df_sous_nutrition['Nombre_personnes'] = df_sous_nutrition['Nombre_personnes'].fillna(0)

In [38]:
df_sous_nutrition['Nombre_personnes'].isnull().sum()

0

In [115]:
# Modifions le type de la colonne 'Nombre_personnes' en float  car les valeurs  sont en str    
df_sous_nutrition['Nombre_personnes'] = df_sous_nutrition['Nombre_personnes'].str.replace('<0.1','0').astype(float)

C:\Users\moust\AppData\Local\Temp/ipykernel_2744/3224682073.py:2: FutureWarning: The default value of regex will change from True to False in a future version.
  df_sous_nutrition['Nombre_personnes'] = df_sous_nutrition['Nombre_personnes'].str.replace('<0.1','0').astype(float)


In [118]:
# Vérifions le nombre de nos valeurs nulles 594 plus les 120 (<0.1)
df_sous_nutrition['Nombre_personnes'].value_counts()

0.0      714
0.3       48
0.4       32
0.2       28
1.5       20
        ... 
3.7        1
24.1       1
189.2      1
190.1      1
8.3        1
Name: Nombre_personnes, Length: 139, dtype: int64

In [119]:
# Vérifions qu'il n y ait pas des valeurs nulles 
df_sous_nutrition['Nombre_personnes'].isnull().sum()

0

In [554]:
# Modifions les valeurs 'Nombre_personnes' en millions d'habitants
df_sous_nutrition['Nombre_personnes'] = df_sous_nutrition['Nombre_personnes']*1000000
df_sous_nutrition['Nombre_personnes']

0       8.600000e+12
1       8.800000e+12
2       8.900000e+12
3       9.700000e+12
4       1.050000e+13
            ...     
1213    0.000000e+00
1214    0.000000e+00
1215    0.000000e+00
1216    0.000000e+00
1217    0.000000e+00
Name: Nombre_personnes, Length: 1218, dtype: float64

In [121]:
# Vérifions le changement du type de données de notre colonne 'Nombre de personnes'
df_sous_nutrition.dtypes

Pays                 object
Année                object
Nombre_personnes    float64
dtype: object

#### II.2.4. Aide alimentaire

In [122]:
# Affichons les 5 premières entrées pour avoir un aperçu de notre dataframe
df_aide_alim.head()

Pays bénéficiaire  Année              Produit  Valeur
0       Afghanistan   2013  Autres non-céréales     682
1       Afghanistan   2014  Autres non-céréales     335
2       Afghanistan   2013         Blé et Farin   39224
3       Afghanistan   2014         Blé et Farin   15160
4       Afghanistan   2013             Céréales   40504

In [123]:
# Affichons le nombre de lignes et de colonnes
df_aide_alim.shape

(1475, 4)

In [124]:
# Vérifions le type de données de notre dataframe
df_aide_alim.dtypes

Pays bénéficiaire    object
Année                 int64
Produit              object
Valeur                int64
dtype: object

In [125]:
# Renommons le titre de la colonne 'valeur' en 'Quantité'
df_aide_alim.rename(columns={'Valeur':'Quantité'},inplace=True)

In [126]:
# Affichons les colonnes de la dataframe
df_aide_alim.columns

Index(['Pays bénéficiaire', 'Année', 'Produit', 'Quantité'], dtype='object')

In [127]:
# Recherchons les valeurs manquantes
df_aide_alim.isnull().sum()

Pays bénéficiaire    0
Année                0
Produit              0
Quantité             0
dtype: int64

In [128]:
# Modifions la colonne 'Quantité' en tonnes
df_aide_alim['Quantité'] = df_aide_alim['Quantité'] * 1000

In [129]:
# Vérifions le changement de ma colonne 'Quantité' en tonnes
df_aide_alim['Quantité'].head(5)

0      682000
1      335000
2    39224000
3    15160000
4    40504000
Name: Quantité, dtype: int64

### Deuxième Partie - Demandes de Marc, Melanie et notes de Julien


### I. Demandes de Marc

#### I.1. Proportion de personnes en état de sous-nutrition en 2017

In [435]:
# Affichons notre dataframe en selectionnant les colonnes necessaires
df_sous_nutrition[['Année','Nombre_personnes']]

Année  Nombre_personnes
0     2012-2014         8600000.0
1     2013-2015         8800000.0
2     2014-2016         8900000.0
3     2015-2017         9700000.0
4     2016-2018        10500000.0
...         ...               ...
1213  2013-2015               0.0
1214  2014-2016               0.0
1215  2015-2017               0.0
1216  2016-2018               0.0
1217  2017-2019               0.0

[1218 rows x 2 columns]

In [436]:
### Modifions l'index dans un premier temps pour mieux localiser notre colonne 'Année'
df_sous_nutrition_prop = df_sous_nutrition.set_index('Année')
df_sous_nutrition_prop

Pays  Nombre_personnes
Année                                   
2012-2014  Afghanistan         8600000.0
2013-2015  Afghanistan         8800000.0
2014-2016  Afghanistan         8900000.0
2015-2017  Afghanistan         9700000.0
2016-2018  Afghanistan        10500000.0
...                ...               ...
2013-2015     Zimbabwe               0.0
2014-2016     Zimbabwe               0.0
2015-2017     Zimbabwe               0.0
2016-2018     Zimbabwe               0.0
2017-2019     Zimbabwe               0.0

[1218 rows x 2 columns]

In [437]:
# Determinons le nombre de personnes en etat de sous_nutrition pour l'année 2017
df_sous_nutrition_prop.loc['2016-2018', 'Nombre_personnes'].sum()

535700000.0

In [438]:
# Filtrons notre population pour l'année 2017
df_population_2017 = df_population[df_population.Année==2017] 

In [547]:
# Determinons le nombre d'habitants  pour l'année 2017
df_population_2017.Nombre_habitants.sum()

7548134097

In [545]:
# Calculons le rapport entre personnes en état de sous_nutrition et la population pour l'année 2017
proportion_personnes_etat_sous_nutrition = round(df_sous_nutrition_prop.loc['2016-2018', 'Nombre_personnes'].sum() * 100 / df_population_2017.Nombre_habitants.sum(),2)
proportion_personnes_etat_sous_nutrition
print(f'La proportion des personnes en état de sous_nutrition dans le monde est de {proportion_personnes_etat_sous_nutrition:.2f}%')

La proportion des personnes en état de sous_nutrition dans le monde est de 7.10%


#### I.2. Le nombre théorique de personnes qui pourraient être nourries en 2017

In [136]:
#Fusionnons nos 2 dataframes sur une colonne commune 'Pays'
dispo_alimentaire = df_dispon_alim.merge(df_population_2017, on='Pays', how='inner')

In [137]:
# Affichage de la variable 'dispo_alimenatire'
dispo_alimentaire

Pays                Produit   Origine  Aliments_Animaux  \
0       Afghanistan       Abats Comestible   animale               0.0   
1       Afghanistan        Agrumes, Autres  vegetale               0.0   
2       Afghanistan  Aliments pour enfants  vegetale               0.0   
3       Afghanistan                 Ananas  vegetale               0.0   
4       Afghanistan                Bananes  vegetale               0.0   
...             ...                    ...       ...               ...   
15411  Îles Salomon       Viande de Suides   animale               0.0   
15412  Îles Salomon    Viande de Volailles   animale               0.0   
15413  Îles Salomon          Viande, Autre   animale               0.0   
15414  Îles Salomon                    Vin  vegetale               0.0   
15415  Îles Salomon         Épices, Autres  vegetale               0.0   

       Autres_utilisations  Dispo_alim_kcal  Dispo_alim_qte  \
0                      0.0              5.0            1.72   
1                      0.0              1.0            1.29   
2                      0.0              1.0            0.06   
3                      0.0              0.0            0.00   
4                      0.0              4.0            2.70   
...                    ...              ...             ...   
15411                  0.0             45.0            4.70   
15412                  0.0             11.0            3.34   
15413                  0.0              0.0            0.06   
15414                  0.0              0.0            0.07   
15415                  0.0              4.0            0.48   

       Dispo_mat_gras_qte  Dispo_prot_qte  Dispo_inter  Export_qte  \
0                    0.20            0.77   53000000.0         0.0   
1                    0.01            0.02   41000000.0   2000000.0   
2                    0.01            0.03    2000000.0         0.0   
3                    0.00            0.00          0.0         0.0   
4                    0.02            0.05   82000000.0         0.0   
...                   ...             ...          ...         ...   
15411                4.28            1.41    3000000.0         0.0   
15412                0.69            1.14    2000000.0         0.0   
15413                0.00            0.04          0.0         0.0   
15414                0.00            0.00          0.0         0.0   
15415                0.21            0.15          0.0         0.0   

       Import_qte  Nourriture     Pertes  Production  Semences  Traitement  \
0             0.0  53000000.0        0.0  53000000.0       0.0         0.0   
1      40000000.0  39000000.0  2000000.0   3000000.0       0.0         0.0   
2       2000000.0   2000000.0        0.0         0.0       0.0         0.0   
3             0.0         0.0        0.0         0.0       0.0         0.0   
4      82000000.0  82000000.0        0.0         0.0       0.0         0.0   
...           ...         ...        ...         ...       ...         ...   
15411         0.0   3000000.0        0.0   2000000.0       0.0         0.0   
15412   2000000.0   2000000.0        0.0         0.0       0.0         0.0   
15413         0.0         0.0        0.0         0.0       0.0         0.0   
15414         0.0         0.0        0.0         0.0       0.0         0.0   
15415         0.0         0.0        0.0         0.0       0.0         0.0   

       Var_stock  Année  Nombre_habitants  
0            0.0   2017          36296113  
1            0.0   2017          36296113  
2            0.0   2017          36296113  
3            0.0   2017          36296113  
4            0.0   2017          36296113  
...          ...    ...               ...  
15411        0.0   2017            636039  
15412        0.0   2017            636039  
15413        0.0   2017            636039  
15414        0.0   2017            636039  
15415        0.0   2017            636039  

[15416 rows x 20 columns]

In [544]:
#calculons la disponibilité alimentaire en kcal pour une année
dispo_alimentaire_kcal = dispo_alimentaire['Dispo_alim_kcal'] * dispo_alimentaire['Nombre_habitants'] * 365
dispo_alimentaire_kcal.sum()
print(f'la disponibilité alimentaire pour une année est de {dispo_alimentaire_kcal.sum():.0f}')

la disponibilité alimentaire pour une année est de 7635429382287555


In [543]:
#calculons le nombre de personnes pouvant être nourries 
personnes_theorique_nourries_2017 = dispo_alimentaire_kcal.sum() / (2500 * 365)
personnes_theorique_nourries_2017
print(f'le nombre de personnes pouvant être nourries en 2017 est de {personnes_theorique_nourries_2017:.0f}')

le nombre de personnes pouvant être nourries en 2017 est de 8367593844


In [542]:
# Calculons le rapport entre le nombre de personnes pouvant être nourries et la population pour l'année 2017
prop_personnes_nourries_2017 = personnes_theorique_nourries_2017 * 100 / df_population_2017.Nombre_habitants.sum()
prop_personnes_nourries_2017
print(f'La proportion des personnes qui pourraient être nourries en 2017 est de {prop_personnes_nourries_2017:.2f}%')

La proportion des personnes qui pourraient être nourries en 2017 est de 110.86%


#### I.3. Nombre de personnes qu'on pourrait nourrir seulement avec l'alimentation végétale

In [142]:
# Filtrons notre dataframe por ne garder que les produits d'origine végétale
dispo_vegetale = dispo_alimentaire[dispo_alimentaire.Origine=='vegetale']
dispo_vegetale

Pays                Produit   Origine  Aliments_Animaux  \
1       Afghanistan        Agrumes, Autres  vegetale               0.0   
2       Afghanistan  Aliments pour enfants  vegetale               0.0   
3       Afghanistan                 Ananas  vegetale               0.0   
4       Afghanistan                Bananes  vegetale               0.0   
6       Afghanistan                  Bière  vegetale               0.0   
...             ...                    ...       ...               ...   
15406  Îles Salomon                 Sésame  vegetale               0.0   
15407  Îles Salomon                    Thé  vegetale               0.0   
15408  Îles Salomon                Tomates  vegetale               0.0   
15414  Îles Salomon                    Vin  vegetale               0.0   
15415  Îles Salomon         Épices, Autres  vegetale               0.0   

       Autres_utilisations  Dispo_alim_kcal  Dispo_alim_qte  \
1                      0.0              1.0            1.29   
2                      0.0              1.0            0.06   
3                      0.0              0.0            0.00   
4                      0.0              4.0            2.70   
6                      0.0              0.0            0.09   
...                    ...              ...             ...   
15406                  0.0              0.0            0.01   
15407                  0.0              0.0            0.17   
15408                  0.0              0.0            0.07   
15414                  0.0              0.0            0.07   
15415                  0.0              4.0            0.48   

       Dispo_mat_gras_qte  Dispo_prot_qte  Dispo_inter  Export_qte  \
1                    0.01            0.02   41000000.0   2000000.0   
2                    0.01            0.03    2000000.0         0.0   
3                    0.00            0.00          0.0         0.0   
4                    0.02            0.05   82000000.0         0.0   
6                    0.00            0.00    3000000.0         0.0   
...                   ...             ...          ...         ...   
15406                0.00            0.00          0.0         0.0   
15407                0.00            0.04          0.0         0.0   
15408                0.00            0.00          0.0         0.0   
15414                0.00            0.00          0.0         0.0   
15415                0.21            0.15          0.0         0.0   

       Import_qte  Nourriture     Pertes  Production  Semences  Traitement  \
1      40000000.0  39000000.0  2000000.0   3000000.0       0.0         0.0   
2       2000000.0   2000000.0        0.0         0.0       0.0         0.0   
3             0.0         0.0        0.0         0.0       0.0         0.0   
4      82000000.0  82000000.0        0.0         0.0       0.0         0.0   
6       3000000.0   3000000.0        0.0         0.0       0.0         0.0   
...           ...         ...        ...         ...       ...         ...   
15406         0.0         0.0        0.0         0.0       0.0         0.0   
15407         0.0         0.0        0.0         0.0       0.0         0.0   
15408         0.0         0.0        0.0         0.0       0.0         0.0   
15414         0.0         0.0        0.0         0.0       0.0         0.0   
15415         0.0         0.0        0.0         0.0       0.0         0.0   

       Var_stock  Année  Nombre_habitants  
1            0.0   2017          36296113  
2            0.0   2017          36296113  
3            0.0   2017          36296113  
4            0.0   2017          36296113  
6            0.0   2017          36296113  
...          ...    ...               ...  
15406        0.0   2017            636039  
15407        0.0   2017            636039  
15408        0.0   2017            636039  
15414        0.0   2017            636039  
15415        0.0   2017            636039  

[11751 rows x 20 columns]

In [541]:
#calculons la disponibilité alimentaire vegetale en kcal pour une année
dispo_alimentaire_vegetale_kcal = dispo_vegetale['Dispo_alim_kcal'] * dispo_vegetale['Nombre_habitants'] * 365
dispo_alimentaire_vegetale_kcal.sum()
print(f'la disponibilité alimentaire végétale pour une année est de {dispo_alimentaire_vegetale_kcal.sum():.0f}')

la disponibilité alimentaire végétale pour une année est de 6300178932059760


In [540]:
#calculons le nombre de personnes pouvant être nourries avec l'alimentation vegetale
personnes_theorique_nourries_vegetale_2017 = dispo_alimentaire_vegetale_kcal.sum() / (2500 * 365)
personnes_theorique_nourries_vegetale_2017
print(f'le nombre de personnes pouvant être nourries avec une alimentation végétale en 2017 est de {personnes_theorique_nourries_vegetale_2017:.0f}')

le nombre de personnes pouvant être nourries avec une alimentation végétale en 2017 est de 6904305679


In [539]:
# Calculons le rapport entre le nombre de personnes pouvant être nourries avec une alimentation végétale et la population pour l'année 2017
prop_personnes_nourries_vegetale_2017 = personnes_theorique_nourries_vegetale_2017 * 100 / df_population_2017.Nombre_habitants.sum()
prop_personnes_nourries_vegetale_2017
print(f'La proportion de personnes pouvant être nourries avec les produits végétaux est de {prop_personnes_nourries_vegetale_2017:.2f}%')

La proportion de personnes pouvant être nourries avec les produits végétaux est de 91.47%


#### I.4. Utilisation de la disponibilité intérieure : part qui est attribuée à l’alimentation animale, aux pertes et à l'alimentation humaine

In [147]:
# Fusionnons nos 2 dataframes sur une colonne commune 'Pays'
dispo_interieure = df_dispon_alim.merge(df_population_2017, on='Pays', how='inner')
dispo_interieure

Pays                Produit   Origine  Aliments_Animaux  \
0       Afghanistan       Abats Comestible   animale               0.0   
1       Afghanistan        Agrumes, Autres  vegetale               0.0   
2       Afghanistan  Aliments pour enfants  vegetale               0.0   
3       Afghanistan                 Ananas  vegetale               0.0   
4       Afghanistan                Bananes  vegetale               0.0   
...             ...                    ...       ...               ...   
15411  Îles Salomon       Viande de Suides   animale               0.0   
15412  Îles Salomon    Viande de Volailles   animale               0.0   
15413  Îles Salomon          Viande, Autre   animale               0.0   
15414  Îles Salomon                    Vin  vegetale               0.0   
15415  Îles Salomon         Épices, Autres  vegetale               0.0   

       Autres_utilisations  Dispo_alim_kcal  Dispo_alim_qte  \
0                      0.0              5.0            1.72   
1                      0.0              1.0            1.29   
2                      0.0              1.0            0.06   
3                      0.0              0.0            0.00   
4                      0.0              4.0            2.70   
...                    ...              ...             ...   
15411                  0.0             45.0            4.70   
15412                  0.0             11.0            3.34   
15413                  0.0              0.0            0.06   
15414                  0.0              0.0            0.07   
15415                  0.0              4.0            0.48   

       Dispo_mat_gras_qte  Dispo_prot_qte  Dispo_inter  Export_qte  \
0                    0.20            0.77   53000000.0         0.0   
1                    0.01            0.02   41000000.0   2000000.0   
2                    0.01            0.03    2000000.0         0.0   
3                    0.00            0.00          0.0         0.0   
4                    0.02            0.05   82000000.0         0.0   
...                   ...             ...          ...         ...   
15411                4.28            1.41    3000000.0         0.0   
15412                0.69            1.14    2000000.0         0.0   
15413                0.00            0.04          0.0         0.0   
15414                0.00            0.00          0.0         0.0   
15415                0.21            0.15          0.0         0.0   

       Import_qte  Nourriture     Pertes  Production  Semences  Traitement  \
0             0.0  53000000.0        0.0  53000000.0       0.0         0.0   
1      40000000.0  39000000.0  2000000.0   3000000.0       0.0         0.0   
2       2000000.0   2000000.0        0.0         0.0       0.0         0.0   
3             0.0         0.0        0.0         0.0       0.0         0.0   
4      82000000.0  82000000.0        0.0         0.0       0.0         0.0   
...           ...         ...        ...         ...       ...         ...   
15411         0.0   3000000.0        0.0   2000000.0       0.0         0.0   
15412   2000000.0   2000000.0        0.0         0.0       0.0         0.0   
15413         0.0         0.0        0.0         0.0       0.0         0.0   
15414         0.0         0.0        0.0         0.0       0.0         0.0   
15415         0.0         0.0        0.0         0.0       0.0         0.0   

       Var_stock  Année  Nombre_habitants  
0            0.0   2017          36296113  
1            0.0   2017          36296113  
2            0.0   2017          36296113  
3            0.0   2017          36296113  
4            0.0   2017          36296113  
...          ...    ...               ...  
15411        0.0   2017            636039  
15412        0.0   2017            636039  
15413        0.0   2017            636039  
15414        0.0   2017            636039  
15415        0.0   2017            636039  

[15416 rows x 20 columns]

 __a. Part attribuée à l'alimentation animale__

In [462]:
# Creation de la variable alimentation animale à partir de la dataframe fusionnée
alimentation_animale = dispo_interieure['Aliments_Animaux'].sum()
alimentation_animale
print(f'lalimentation animale  = {alimentation_animale:.0f}')

lalimentation animale  = 1288002000000


In [460]:
# Calculons le rapport entre l'alimentation animale et la disponibilité intérieure
part_alimentation_animale_dispo_interieure = alimentation_animale * 100 / dispo_interieure['Dispo_inter'].sum()
part_alimentation_animale_dispo_interieure
print(f'La part attribuée à lalimentation animale = {part_alimentation_animale_dispo_interieure:.2f}%')

La part attribuée à lalimentation animale = 13.23%


 __b. Part attribuée aux pertes__

In [466]:
# Creation de la variable Pertes à partir de la dataframe fusionnée
pertes = dispo_interieure['Pertes'].sum()
pertes
print(f'Pertes = {pertes:.0f}')

Pertes = 452283000000


In [464]:
# Calculons le rapport entre les pertes et la disponibilité intérieure
part_pertes_dispo_interieure = pertes * 100 / dispo_interieure['Dispo_inter'].sum()
part_pertes_dispo_interieure
print(f'La part attribuée aux pertes = {part_pertes_dispo_interieure:.2f}%')

La part attribuée aux pertes = 4.65%


 __c. Part attribuée à l'alimentation humaine__

In [465]:
# Creation de la variable Alimentation humaine à partir de la dataframe fusionnée
alimentation_humaine = dispo_interieure['Nourriture'].sum()
alimentation_humaine
print(f'Alimentation humaine = {alimentation_humaine:.0f}')

Alimentation humaine = 4805525000000


In [467]:
# Calculons le rapport entre l'alimentation humaine et la disponibilité intérieure
part_alimentation_humaine_dispo_interieure = alimentation_humaine * 100 / dispo_interieure['Dispo_inter'].sum()
part_alimentation_humaine_dispo_interieure
print(f'La part attribuée à lalimentation humaine = {part_alimentation_humaine_dispo_interieure:.2f}%')

La part attribuée à lalimentation humaine = 49.37%


### II. Demandes de Melanie

#### II.1. Pays pour lesquels la proportion de personnes sous-alimentées est la plus forte en 2017

In [178]:
# Fusionnons nos 2 dataframes sur une colonne commune 'Pays'
personnes_sous_alimentées = df_sous_nutrition_2017.merge(df_population_2017, on='Pays', how='inner')
personnes_sous_alimentées

Pays    Année_x  Nombre_personnes  \
0                               Afghanistan  2016-2018        10500000.0   
1                            Afrique du Sud  2016-2018         3100000.0   
2                                   Albanie  2016-2018          100000.0   
3                                   Algérie  2016-2018         1300000.0   
4                                 Allemagne  2016-2018               0.0   
..                                      ...        ...               ...   
198  Venezuela (République bolivarienne du)  2016-2018         8000000.0   
199                                Viet Nam  2016-2018         6500000.0   
200                                   Yémen  2016-2018               0.0   
201                                  Zambie  2016-2018               0.0   
202                                Zimbabwe  2016-2018               0.0   

     Année_y  Nombre_habitants  
0       2017          36296113  
1       2017          57009756  
2       2017           2884169  
3       2017          41389189  
4       2017          82658409  
..       ...               ...  
198     2017          29402484  
199     2017          94600648  
200     2017          27834819  
201     2017          16853599  
202     2017          14236595  

[203 rows x 5 columns]

In [468]:
# Ajoutons la colonne 'proportion des personnes sous_alimentées' et calculons la proportion des personnes sous-alimentées
personnes_sous_alimentées['proportion_personnes_sous_alimentées'] = personnes_sous_alimentées['Nombre_personnes']*100 / personnes_sous_alimentées['Nombre_habitants']
personnes_sous_alimentées

Pays    Année_x  Nombre_personnes  \
0                               Afghanistan  2016-2018        10500000.0   
1                            Afrique du Sud  2016-2018         3100000.0   
2                                   Albanie  2016-2018          100000.0   
3                                   Algérie  2016-2018         1300000.0   
4                                 Allemagne  2016-2018               0.0   
..                                      ...        ...               ...   
198  Venezuela (République bolivarienne du)  2016-2018         8000000.0   
199                                Viet Nam  2016-2018         6500000.0   
200                                   Yémen  2016-2018               0.0   
201                                  Zambie  2016-2018               0.0   
202                                Zimbabwe  2016-2018               0.0   

     Année_y  Nombre_habitants  proportion_personnes_sous_alimentées  
0       2017          36296113                             28.928718  
1       2017          57009756                              5.437666  
2       2017           2884169                              3.467203  
3       2017          41389189                              3.140917  
4       2017          82658409                              0.000000  
..       ...               ...                                   ...  
198     2017          29402484                             27.208586  
199     2017          94600648                              6.870989  
200     2017          27834819                              0.000000  
201     2017          16853599                              0.000000  
202     2017          14236595                              0.000000  

[203 rows x 6 columns]

In [431]:
# Trions les pays ayant les proportions les plus élevées en 2017
proportion_personnes_sous_alimentées_pays_2017 = personnes_sous_alimentées[['Pays', 'proportion_personnes_sous_alimentées']].sort_values('proportion_personnes_sous_alimentées', ascending=False).head(10)
proportion_personnes_sous_alimentées_pays_2017

Pays  \
78                                        Haïti   
157  République populaire démocratique de Corée   
108                                  Madagascar   
103                                     Libéria   
100                                     Lesotho   
183                                       Tchad   
161                                      Rwanda   
121                                  Mozambique   
186                                 Timor-Leste   
0                                   Afghanistan   

     proportion_personnes_sous_alimentées  
78                              48.259182  
157                             47.188685  
108                             41.062924  
103                             38.279742  
100                             38.249438  
183                             37.957606  
161                             35.055619  
121                             32.810898  
186                             32.173531  
0                               28.928718

En 2017, Haïti, la republique populaire démocratique de corée et Madagascar forment le top 3 des pays qui renferment le plus de personnes souffrant de sous-nutrition.

#### II.2. Pays ayant reçus le plus d'aides depuis 2013

In [224]:
# visualisons notre dataframe 'Aide-alimentaire'
df_aide_alim

Pays bénéficiaire  Année                   Produit  Quantité
0          Afghanistan   2013       Autres non-céréales    682000
1          Afghanistan   2014       Autres non-céréales    335000
2          Afghanistan   2013              Blé et Farin  39224000
3          Afghanistan   2014              Blé et Farin  15160000
4          Afghanistan   2013                  Céréales  40504000
...                ...    ...                       ...       ...
1470          Zimbabwe   2015  Mélanges et préparations     96000
1471          Zimbabwe   2013              Non-céréales   5022000
1472          Zimbabwe   2014              Non-céréales   2310000
1473          Zimbabwe   2015              Non-céréales    306000
1474          Zimbabwe   2013                Riz, total     64000

[1475 rows x 4 columns]

In [270]:
# Trions les pays ayant reçus les plus d'aides depuis 2013
df_aide_alim.groupby(['Pays bénéficiaire'])['Quantité'].sum().sort_values(ascending=False).head(10)

Pays bénéficiaire
République arabe syrienne           1858943000
Éthiopie                            1381294000
Yémen                               1206484000
Soudan du Sud                        695248000
Soudan                               669784000
Kenya                                552836000
Bangladesh                           348188000
Somalie                              292678000
République démocratique du Congo     288502000
Niger                                276344000
Name: Quantité, dtype: int64

Depuis 2013, les 3 premiers pays ayant reçus le plus d'aides alimentaires sont la République arabe syrienne, l'Ethiopie, le Yémen avec plus d'un milliard de tonnes 

#### II.3. Pays ayant le plus/le moins de disponibilité/habitant

In [551]:
# Trions les pays ayant le moins de disponibilité alimentaire
df_dispon_alim.groupby(['Pays'])['Dispo_alim_kcal'].sum().sort_values(ascending=True).head(10)

Pays
République centrafricaine                     1879.0
Zambie                                        1924.0
Madagascar                                    2056.0
Afghanistan                                   2087.0
Haïti                                         2089.0
République populaire démocratique de Corée    2093.0
Tchad                                         2109.0
Zimbabwe                                      2113.0
Ouganda                                       2126.0
Timor-Leste                                   2129.0
Name: Dispo_alim_kcal, dtype: float64

Les besoins alimentaires d'un individu étant évalués à 2.500 kcal par jour en moyenne, les habitants des pays dont la disponibilité alimentaire par habitant est le moins élevé ont un apport calorique jour ne permettant pas de couvrir les besoins en énergie.

In [549]:
# Trions les pays ayant le plus de disponibilité alimentaire
df_dispon_alim.groupby(['Pays'])['Dispo_alim_kcal'].sum().sort_values(ascending=False).head(10)

Pays
Autriche                 3770.0
Belgique                 3737.0
Turquie                  3708.0
États-Unis d'Amérique    3682.0
Israël                   3610.0
Irlande                  3602.0
Italie                   3578.0
Luxembourg               3540.0
Égypte                   3518.0
Allemagne                3503.0
Name: Dispo_alim_kcal, dtype: float64

Les besoins alimentaires d'un individu étant évalués à 2.500 kcal par jour en moyenne, les habitants des pays dont la disponibilité alimentaire par habitant est le plus élevé ont un apport calorique jour permettant de couvrir les besoins en énergie.

### III. Notes de Julien

#### III.1. Travaux préalables

In [500]:
# Créeons une liste de cereales à partir du bilan alimentaire de la FAO
liste_cereales = ['Blé','Riz (Eq Blanchi)','Orge','Maïs','Millet','Céréales, Autres','Seigle','Avoine','Sorgho']
liste_cereales

['Blé',
 'Riz (Eq Blanchi)',
 'Orge',
 'Maïs',
 'Millet',
 'Céréales, Autres',
 'Seigle',
 'Avoine',
 'Sorgho']

In [525]:
#  Générons notre dataframe en filtant avec la condition isin
df_cereales = df_dispon_alim[df_dispon_alim["Produit"].isin(liste_cereales)]
df_cereales["Produit"].unique()

array(['Blé', 'Céréales, Autres', 'Maïs', 'Millet', 'Orge',
       'Riz (Eq Blanchi)', 'Avoine', 'Seigle', 'Sorgho'], dtype=object)

In [526]:
# Affichons notre nouveau dataframe 
df_cereales

Pays           Produit   Origine  Aliments_Animaux  \
7       Afghanistan               Blé  vegetale               0.0   
12      Afghanistan  Céréales, Autres  vegetale               0.0   
32      Afghanistan              Maïs  vegetale       200000000.0   
34      Afghanistan            Millet  vegetale               0.0   
40      Afghanistan              Orge  vegetale       360000000.0   
...             ...               ...       ...               ...   
15545  Îles Salomon  Céréales, Autres  vegetale               0.0   
15568  Îles Salomon              Maïs  vegetale               0.0   
15575  Îles Salomon              Orge  vegetale               0.0   
15591  Îles Salomon  Riz (Eq Blanchi)  vegetale               0.0   
15593  Îles Salomon            Sorgho  vegetale               0.0   

       Autres_utilisations  Dispo_alim_kcal  Dispo_alim_qte  \
7                      0.0           1369.0          160.23   
12                     0.0              0.0            0.00   
32                     0.0             21.0            2.50   
34                     0.0              3.0            0.40   
40                     0.0             26.0            2.92   
...                    ...              ...             ...   
15545                  0.0              0.0            0.00   
15568                  0.0              1.0            0.15   
15575                  0.0              0.0            0.07   
15591           12000000.0            623.0           63.76   
15593                  0.0              0.0            0.00   

       Dispo_mat_gras_qte  Dispo_prot_qte   Dispo_inter  Export_qte  \
7                    4.69           36.91  5.992000e+09         0.0   
12                   0.00            0.00  0.000000e+00         0.0   
32                   0.30            0.56  3.130000e+08         0.0   
34                   0.02            0.08  1.300000e+07         0.0   
40                   0.24            0.79  5.240000e+08         0.0   
...                   ...             ...           ...         ...   
15545                0.00            0.00  0.000000e+00         0.0   
15568                0.01            0.03  0.000000e+00         0.0   
15575                0.00            0.01  1.000000e+06         0.0   
15591                1.36           10.90  4.900000e+07         0.0   
15593                0.00            0.00  0.000000e+00         0.0   

         Import_qte    Nourriture       Pertes    Production     Semences  \
7      1.173000e+09  4.895000e+09  775000000.0  5.169000e+09  322000000.0   
12     0.000000e+00  0.000000e+00          0.0  0.000000e+00          0.0   
32     1.000000e+06  7.600000e+07   31000000.0  3.120000e+08    5000000.0   
34     0.000000e+00  1.200000e+07    1000000.0  1.300000e+07          0.0   
40     1.000000e+07  8.900000e+07   52000000.0  5.140000e+08   22000000.0   
...             ...           ...          ...           ...          ...   
15545  0.000000e+00  0.000000e+00          0.0  0.000000e+00          0.0   
15568  0.000000e+00  0.000000e+00          0.0  0.000000e+00          0.0   
15575  1.000000e+06  0.000000e+00          0.0  0.000000e+00          0.0   
15591  4.700000e+07  3.600000e+07    1000000.0  3.000000e+06          0.0   
15593  0.000000e+00  0.000000e+00          0.0  0.000000e+00          0.0   

       Traitement  Var_stock  
7             0.0     -350.0  
12            0.0        0.0  
32            0.0        0.0  
34            0.0        0.0  
40            0.0        0.0  
...           ...        ...  
15545         0.0        0.0  
15568         0.0        0.0  
15575   1000000.0        0.0  
15591         0.0        0.0  
15593         0.0        0.0  

[1497 rows x 18 columns]

#### III.2. Calcul de la proportion des céréales utilisée dans l'alimentation humaine

In [528]:
# Faisons le rapport entre la nourriture et la disponibilité intérieure
proportion_cereales_alimentation_humaine = df_cereales['Nourriture'].sum() * 100 / df_cereales['Dispo_inter'].sum()
proportion_cereales_alimentation_humaine 
print(f'La proportion des céréales dans lalimentation humaine est de {proportion_cereales_alimentation_humaine:.2f}%')

La proportion des céréales dans lalimentation humaine est de 42.75%


#### III.3. Calcul de la proportion des céréales utilisée dans l'alimentation animale

In [530]:
# Faisons le rapport entre les aliments pour animaux et la disponibilité intérieure
proportion_cereales_alimentation_animale = df_cereales['Aliments_Animaux'].sum() * 100 / df_cereales['Dispo_inter'].sum()
proportion_cereales_alimentation_animale
print(f'La proportion des céréales dans lalimentation animale est de {proportion_cereales_alimentation_animale:.2f}%')

La proportion des céréales dans lalimentation animale est de 36.29%


#### III.4. Utilisation du manioc par la Thaïlande aux égards de la proportion de personnes en sous-nutrition

In [379]:
# visualisons notre dataframe 'Population de 2017'
df_population_2017

Pays  Année  Nombre_habitants
4                                Afghanistan   2017          36296113
10                            Afrique du Sud   2017          57009756
16                                   Albanie   2017           2884169
22                                   Algérie   2017          41389189
28                                 Allemagne   2017          82658409
...                                      ...    ...               ...
1390  Venezuela (République bolivarienne du)   2017          29402484
1396                                Viet Nam   2017          94600648
1402                                   Yémen   2017          27834819
1408                                  Zambie   2017          16853599
1414                                Zimbabwe   2017          14236595

[236 rows x 3 columns]

In [413]:
# filtrons la population de thailande sur l'année 2017
population_thailande_2017 = df_population_2017[df_population_2017.Pays=='Thaïlande']
population_thailande_2017

Pays  Année  Nombre_habitants
1312  Thaïlande   2017          69209810

In [534]:
# Affichons la population de la Thailande en 2017
nombre_habitants_thailande_2017 = population_thailande_2017['Nombre_habitants'].sum()
nombre_habitants_thailande_2017
print(f'Le nombre de Thaïlandais en 2017 est de {nombre_habitants_thailande_2017:}')

Le nombre de Thaïlandais en 2017 est de 69209810


In [418]:
# visualisons notre dataframe 'Sous-nutrition de 2017'
df_sous_nutrition_2017

Pays      Année  Nombre_personnes
4                                Afghanistan  2016-2018        10500000.0
10                            Afrique du Sud  2016-2018         3100000.0
16                                   Albanie  2016-2018          100000.0
22                                   Algérie  2016-2018         1300000.0
28                                 Allemagne  2016-2018               0.0
...                                      ...        ...               ...
1192  Venezuela (République bolivarienne du)  2016-2018         8000000.0
1198                                Viet Nam  2016-2018         6500000.0
1204                                   Yémen  2016-2018               0.0
1210                                  Zambie  2016-2018               0.0
1216                                Zimbabwe  2016-2018               0.0

[203 rows x 3 columns]

In [421]:
# filtrons le nombre de personnes en sous_nutrition en thailande sur l'année 2017
sous_nutrition_thailande_2017 = df_sous_nutrition_2017[df_sous_nutrition_2017.Pays=='Thaïlande']
sous_nutrition_thailande_2017

Pays      Année  Nombre_personnes
1114  Thaïlande  2016-2018         6200000.0

In [538]:
# Affichons le nombre de personnes en état de sous-nutrition en Thaïlande en 2017
nombre_personnes_thailande_2017 =sous_nutrition_thailande_2017['Nombre_personnes'].sum()
nombre_personnes_thailande_2017
print(f'Le nombre de Thaïlandais en état de sous-nutrition en 2017 est de {nombre_personnes_thailande_2017:.0f}')

Le nombre de Thaïlandais en état de sous-nutrition en 2017 est de 6200000


In [524]:
# calculons la proportion des personnes en état de sous-nutrition en Thaïlande
proportion_personnes_sous_nutrition_thailande = nombre_personnes_thailande_2017 * 100 / nombre_habitants_thailande_2017 
proportion_personnes_sous_nutrition_thailande
print(f'La proportion des personnes en état de sous-nutrition en Thailande est de {proportion_personnes_sous_nutrition_thailande:.2f}%')

La proportion des personnes en état de sous-nutrition en Thailande est de 8.96%


__d. Rapport entre l'exportation et la production du manioc__

In [522]:
# Calculons de la valeur de la production du manioc thailandais
production_thailande = df_dispon_alim[(df_dispon_alim.Pays=='Thaïlande')&(df_dispon_alim.Produit=='Manioc')]['Production'].sum()
production_thailande
print(f' la Thaïlande produit {production_thailande:.0f} tonnes de manioc')

 la Thaïlande produit 30228000000 tonnes de manioc


In [521]:
# Calculons de la valeur de l'exportation du manioc thailandais
exportation_thailande = df_dispon_alim[(df_dispon_alim.Pays=='Thaïlande')&(df_dispon_alim.Produit=='Manioc')]['Export_qte'].sum()
exportation_thailande
print(f' la Thaïlande exporte {exportation_thailande:.0f} tonnes de manioc')

 la Thaïlande exporte 25214000000 tonnes de manioc


In [523]:
# Calculons de la part de l'exportation dans la production du manioc thailandais
proportion_manioc_thailande = exportation_thailande * 100 / production_thailande
proportion_manioc_thailande
print(f'La part de lexportation dans la production du manioc = {proportion_manioc_thailande:.2f}%')

La part de lexportation dans la production du manioc = 83.41%


### IV. Conclusions

- En 2017 sur une population de 7548134097 d'habitants, 7 personnes sur 100 souffraient de sous-alimentation dans le monde.

- Avec la disponibilité alimentaire globale on pourrait nourrir 8367593844 personnes, c’est à dire le 110.86% de la population mondiale en 2017.

- Les seuls produits végétaux pourraient nourrir 6904305679 personnes, c’est à dire le 91.47% de la population mondiale en 2017.

- L’alimentation humaine représente pas loin de la moitié de la disponibilité intérieure 49,37 % avec une perte pas négligeable de 4,65 % observée (Les pertes pouvant être dues à un accident météorologique (pluie, tempête...), à la prédation ou la dégradation par la faune sauvage, les prédateurs(oiseaux, sangliers) et au mauvais réglage des machines ou à des machines de mauvaises qualités…

- En 2017, Haïti, la république populaire démocratique de corée et Madagascar forment le top 3 des pays qui comprennent le plus de personnes souffrant de sous-nutrition. 

- Depuis 2013, les 3 premiers pays ayant reçus le plus d'aides alimentaires sont la République arabe syrienne, l’Éthiopie, le Yémen avec plus d'un milliard de tonnes. Il est à noter que l'origine de l'aide alimentaire vient de l'écoulement des excédents alimentaires. Le risque de ce type d'aide est de supplanter l'importation commerciale et de décourager la production locale dans les pays bénéficiaires. Les pays pauvres n’ont souvent pas les infrastructures de stockages nécessaires pour recevoir de telles donations, et encore moins pour les distribuer. Il vaut mieux transférer l'expertise agricole car les aides peuvent augmenter la précarité de leur condition en les faisant négliger les cultures locales. Les dons peuvent entraîner la perte des compétences agricoles.

- Les besoins alimentaires d'un individu étant évalués à 2.500 kcal par jour en moyenne, les habitants des pays dont la disponibilité alimentaire par habitant  est le moins élevé ont un apport calorique jour ne permettant pas de couvrir les besoins en énergie. La République centrafricaine  avec ses 1879 calories par habitant arrive première de la liste. Concernant les pays dont la disponibilité alimentaire par habitant est le plus élevé l’Autriche en tête avec 3770 kcal par habitant suivi de la Belgique avec 3737 kcal par habitants. La plupart de ces pays sont européens contrairement à la première liste qui comprenait principalement les pays africains, et d’Asie. L'auto-suffisance alimentaire des pays europeensDans les pays riches, le type d'agriculture est dit intensif. Il s'agit d'une agriculture productiviste, en grande partie tournée vers l'exportation qui permet d'une part de nourrir la population locale et d'autre part, d'exporter vers d'autres pays une grosse partie de la production. Pour parvenir à ce résultat, les paysans utilisent massivement des engrais chimiques, des traitements herbicides, des insecticides, des pesticides, des régulateurs de croissance... et s'équipent d'un matériel agricole sophistiqué . Grâce à cela, les rendements et les récoltes sont bien meilleurs que dans les pays en développement où l'agriculture est, le plus souvent, de type traditionnel. Les paysans sont trop pauvres pour parvenir à s'acheter des outils, des machines, des engrais, des pesticides et des bonnes semences. 

- Plus d’un tiers de la production mondiale en céréales est destinée a l’alimentation animale soit 36,29 %. 

- La proportion des personnes en état de sous-nutrition en Thaïlande est de 8.96%. Et pourtant, l'exportation du manioc thailandais constitue 83% de la production du pays. 
La Thaïlande pourrait résoudre le problème de la sous-nutrition en réduisant l’exportation du manioc et en le transformant afin de combler la demande intérieure. Même si le manioc à lui seul ne saurait résoudre le problème.